packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts

```diff
@@ -261,14 +290,21 @@ export class GraphQLSitemapService {
           )
         );
       }
+
+      if (!item.route?.personalization?.variantIds.length && siteName) {
```

Comment:
At this point we should have all the paths, but without the site prefix in aggregatedPaths. We can transform them directly if needed.
Instead of pushing more paths, it's probably easier to just run a for loop on them to transform existing paths to have a site prefix when multisite is on.
Or, alternatively, apply site transformation when we add paths on line 287 and earlier (via formatPath perhaps)

Reply:
too many paths :cold_sweat:

Reply:
i have refactored the function. :)